src-cpp/rdkafkacpp.h

```diff
@@ -577,7 +578,8 @@ class RD_EXPORT OAuthBearerTokenRefreshCb {
   /**
    * @brief SASL/OAUTHBEARER token refresh callback class.
    */
-  virtual void oauthbearer_token_refresh_cb (const std::string &oauthbearer_config) = 0;
+  virtual void oauthbearer_token_refresh_cb (RdKafka::Handle* handle,
```

Comment:
This is a breaking change that is not allowed by our API compatibility guarantees.
The old method needs to stay in place (but can be deprecated so we can remove it in v2).

Reply:
Ah, right...

How about something like this:
* `virtual void oauthbearer_token_refresh_cb(const std::string&) = 0;` remains intact
* Add `virtual void oauthbearer_token_refresh_cb(RdKafka::Handle*, const std::string&);` with a default implementation that simply forwards to `oauthbearer_token_refresh_cb(const std::string&)`

This would allow for an implementation of this interface the option of using the newer API, while requiring no changes from existing implementations. Because the original API is pure virtual, you'd still have to supply an implementation for `oauthbearer_token_refresh_cb(const std::string &)` until it is removed, though it can be empty if using the newer API.

Thoughts?

Reply:
That makes sense to me; it would become `virtual void oauthbearer_token_refresh_cb(RdKafka::Handle*, const std::string&) = 0;` when the deprecated method is removed (i.e. in v2).

Reply:
Reiterating on this I'm leaning towards actually breaking the API just this once since the proposed fix here will break existing applications that make use of this interface anyway, right?

And I don't think many people are using the C++ OAUTHBEARER interface.

Thoughts?

Reply:
The changes I proposed on Nov 12th would allow applications currently using this interface to remain unchanged with the additional method. Once the deprecated method is removed in favor of the additional one, they'll break.

I don't have strong opinions on whether it's best to "fix" it now or later though, I'm okay with either.

Reply:
After a week or so, I'm more and more of the opinion that this is a reasonable breaking change to make if you're okay with it.

Reply:
@rondagostino What is your take on this?
We want to break the current C++ API, which is a semver sin, but there are two limiting factors:
 1. very few people, if any, are using the C++ OAUTHBEARER interface.
 2. We don't provide any ABI guarantees for the C++ library, so applications need to be recompiled anyway on upgrade, and this is a small fix for the developer to make.

Reply:
@edenhill I think that is an accurate summary of the situation.  I'm disappointed that this problem exists -- I wish I had realized it back when I did this. :-(

There really isn't a great solution other than bumping the major version and fixing it/breaking existing stuff, but if that isn't possible right now then I say just go for it and break things now without bumping the major version as opposed to jumping through the hoop imposed by semantic versioning making a messy situation messier.

Reply:
Thanks @rondagostino .

@chrisbeard Do you want to update the PR with the outcome of this discussion, namely that we'll break API backwards compat and change the existing method signature?

Reply:
Hey @edenhill, finally got back to this. I've reverted back to the non backwards compatible interface that passes the `RdKafka::Handle*` to the callback.

Let me know if there's anything that needs cleaning up. 0105_transactions_mock seems to be failing, but seems (hopefully) unrelated?